### PR TITLE
framework: Update build

### PR DESCRIPTION
**Pull request by mestery**
This moves back to the somewhat byzantine way that sessionOffload is currently built. It does, however, update some components. Instead of the since abandoned (pour one out) CentOS, we move to Rocky Linux. We also update the versions of cmake, grpc, and libconfig.

Signed-off-by: Kyle Mestery <mestery@mestery.com>

### CODE REVIEW

File: openoffload/cpp/framework/build/Dockerfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-FROM centos:8.4.2105
+FROM rockylinux:9
```
Comment (glimchb) [inline openoffload/cpp/framework/build/Dockerfile:1]: I would use fedora... like we do in other repos...

**Review by glimchb — APPROVED** [review]
reasonable. small nit abour fedora vs rocky...

File: openoffload/cpp/framework/build/Dockerfile (hunk 1 above)
Comment (mestery) [reply openoffload/cpp/framework/build/Dockerfile:1]: Fedora isn't CentOS, Rocky Linux is what CentOS used to be, so I will ignore this comment. See https://rockylinux.org

Comment (seroyer) [reply openoffload/cpp/framework/build/Dockerfile:1]: Or `quay.io/centos/centos:stream9`

**Pull request merged**